File: python/leetcode/_I_1607_E.py

class Solution:
    def maximum(self, a: int, b: int) -> int:
        sub = a - b
        sign = sub >> 63
        return (1 + sign) * a + (-sign) * b


if __name__ == '__main__':
    print(Solution().maximum(2147483647, -2147483648))

File: c++/src/leetcode/_I_1607_E.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int maximum(int a, int b) {
        long sub = (long) a - b;
        int sign = sub >> 63;
        return (1 + sign) * a + (-sign) * b;
    }
};

int main() {
    Solution solution;
    cout << solution.maximum(2147483647, -2147483648) << endl;
}
